### PR TITLE
Bug fixed while reading closed response body

### PR DESCRIPTION
**Pull request by mirshahriar**
#### Summary

Fixed a bug that occurred while reading on the closed response body

#### Ticket Link

[Jira Ticket](https://mattermost.atlassian.net/browse/MM-41250)

#### Release Note

```release-note
Fixed a bug that occurred while reading on the closed response body
```

### CODE REVIEW

File: model/client.go [hunk 1]
```diff
@@ -306,36 +307,54 @@ func (c *Client) doDelete(u string) (*http.Response, error) {
...
+	archiveBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", errors.New("failed to read response body")
+	}
+
+	if archiveBytes == nil {
+		return "", errors.New("invalid response data")
+	}
+
+	return string(archiveBytes), nil
```
Comment (fmartingr) [inline model/client.go:357]: nit: if this is archive name (not contents) I would rename the variable to `archiveNameBytes` or something similar. I got confused between this name and the `archiveID` above.

**Review by fmartingr — APPROVED** [review]
LGTM! Kudos for updating the README as well

File: model/client.go (hunk 1 above)
Comment (mirshahriar) [reply model/client.go]: Thank you.  It was really confusing. Fixed

**Review by gabrieljackson — APPROVED** [review]
👍

**Pull request merged**